### PR TITLE
[opt](inverted index) performance optimization for need_read_data in compound

### PR DESCRIPTION
**Pull request by zzzxl1993**
…the compound process

## Proposed changes

1. performance is relatively low when there are too many query conditions

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by zzzxl1993** [comment]
run buildall

**Review by xiaokang — DISMISSED** [review]
LGTM

**Comment by zzzxl1993** [comment]
run buildall

**Comment by zzzxl1993** [comment]
run buildall

**Review by xiaokang — DISMISSED** [review]
LGTM

File: be/src/olap/rowset/segment_v2/segment_iterator.cpp [hunk 1]
```diff
@@ -933,18 +936,9 @@ Status SegmentIterator::_apply_index_except_leafnode_of_andnode() {
...
-        auto column_name = _schema->column(pred->column_id())->name();
-        if (!_remaining_conjunct_roots.empty() &&
-            _check_column_pred_all_push_down(column_name, true,
-                                             pred->type() == PredicateType::MATCH) &&
-            !pred->predicate_params()->marked_by_runtime_filter) {
-            // if column's need_read_data already set true, we can not set it to false now.
-            if (_need_read_data_indices.find(pred->column_id()) == _need_read_data_indices.end()) {
-                _need_read_data_indices[pred->column_id()] = false;
-            }
+        if (!_need_read_data_indices.contains(column_id)) {
```
Comment (airborne12) [inline be/src/olap/rowset/segment_v2/segment_iterator.cpp:940]: lost condition "!pred->predicate_params()->marked_by_runtime_filter" ?

**Comment by zzzxl1993** [comment]
run buildall

**Review by xiaokang — APPROVED** [review]
LGTM

**Review by xiaokang — APPROVED** [review]
LGTM

**Review by airborne12 — APPROVED** [review]
LGTM

**Review by qidaye — APPROVED** [review]

**Pull request merged**